inc/environment_indicator/namespace.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Altis\Cloud\Environment_Indicator;
+
+use function Altis\get_environment_name;
+use function Altis\get_environment_type;
+use WP_Admin_Bar;
+
+function bootstrap() {
+	add_action( 'admin_bar_init', __NAMESPACE__ . '\\enqueue_admin_scripts' );
+	// Allow other modules to add sub-menu items to Altis admin bar menu first.
+	add_action( 'admin_bar_menu', __NAMESPACE__ . '\\add_admin_bar_env_info', 15 );
+}
+
+/**
+ * Add menu item to admin bar to display environment info:
+ * - environment type
+ * - environment name
+ * - URL to Altis dashboard for the environment
+ *
+ * @param WP_Admin_Bar $wp_admin_bar WP_Admin_Bar instance, passed by reference.
+ */
+function add_admin_bar_env_info( WP_Admin_Bar $wp_admin_bar ) {
+	if ( ! is_admin_bar_showing() ) {
+		return;
+	}
+
+	// Specify environments for which to show indicator in the admin bar.
+	$show_indicator = in_array( get_environment_type(), [ 'local', 'development', 'staging' ], true );
+	$show_indicator = apply_filters( 'altis.show_environment_indicator', $show_indicator, get_environment_type() );
+	if ( ! $show_indicator ) {
+		return;
+	}
+
+	// Environment types text to be displayed in the admin bar.
+	$envs = [
+		'local'       => esc_html_x( 'local', 'Server environment type', 'altis' ),
+		'development' => sprintf( '<abbr title="%s">%s</abbr>',
+			esc_attr_x( 'development', 'Server environment type - full form', 'altis' ),
+			esc_html_x( 'dev', 'Server environment type - abbreviated', 'altis' )
+		),
+		'staging'     => esc_html_x( 'staging', 'Server environment type', 'altis' ),
+		'production'  => sprintf( '<abbr title="%s">%s</abbr>',
+			esc_attr_x( 'production', 'Server environment type - full form', 'altis' ),
+			esc_html_x( 'prod', 'Server environment type - abbreviated', 'altis' )
+		),
+	];
```

Comment:
@roborourke I went ahead and used the `esc` functions here as it's most straightforward.
However, I know we should escape late.. should I instead do this on line https://github.com/humanmade/altis-cloud/pull/126/files#diff-dbc927f3487c6469f372e472b9e3cbdcR52 and use `whitelist_html()` to list allowed HTML?

Reply:
I think this is fine tbh, nicely done